cluster-autoscaler/simulator/csi/snapshot/snapshot.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshot
+
+import (
+	"fmt"
+
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/common"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/framework"
+	fwk "k8s.io/kube-scheduler/framework"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// Snapshot represents a snapshot of CSI node information for cluster simulation.
+type Snapshot struct {
+	csiNodes *common.PatchSet[string, *storagev1.CSINode]
+}
+
+// NewSnapshot creates a new Snapshot from a map of CSI nodes.
+func NewSnapshot(csiNodes map[string]*storagev1.CSINode) *Snapshot {
+	csiNdodePatch := common.NewPatchFromMap(csiNodes)
+	return &Snapshot{
+		csiNodes: common.NewPatchSet(csiNdodePatch),
+	}
+}
+
+func (s *Snapshot) listCSINodes() []*storagev1.CSINode {
+	csiNodes := s.csiNodes.AsMap()
+	csiNodesList := make([]*storagev1.CSINode, 0, len(csiNodes))
+	for _, csiNode := range csiNodes {
+		csiNodesList = append(csiNodesList, csiNode)
+	}
+	return csiNodesList
+}
+
+// CSINodes returns a CSI node lister for the snapshot.
+func (s *Snapshot) CSINodes() fwk.CSINodeLister {
+	return SnapshotCSINodeLister{snapshot: s}
+}
+
+// AddCSINodes adds a list of CSI nodes to the snapshot.
+func (s *Snapshot) AddCSINodes(csiNodes []*storagev1.CSINode) error {
+	for _, csiNode := range csiNodes {
+		err := s.AddCSINode(csiNode)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Get returns a CSI node from the snapshot by name.
+func (s *Snapshot) Get(name string) (*storagev1.CSINode, error) {
+	csiNode, found := s.csiNodes.FindValue(name)
+	if !found {
+		return nil, fmt.Errorf("csi nodes %s not found", name)
+	}
+	return csiNode, nil
+}
+
+// WrapSchedulerNodeInfo wraps a scheduler node info with a CSI node.
+func (s *Snapshot) WrapSchedulerNodeInfo(schedNodeInfo *schedulerframework.NodeInfo) (*framework.NodeInfo, error) {
```

Comment:
This isn't actually used anywhere, right? Can you remove it?

Reply:
removed.